src/cmap/conn/wire/message.rs

```diff
@@ -84,12 +91,108 @@ impl Message {
         let mut reader = BufReader::new(reader);
         let header = Header::read_from(&mut reader).await?;
 
+        if header.op_code == OpCode::Message {
+            return Self::read_from_op_msg(reader, &header).await;
+        }
+        #[cfg(any(
+            feature = "zstd-compression",
+            feature = "zlib-compression",
+            feature = "snappy-compression"
+        ))]
+        if header.op_code == OpCode::Compressed {
+            return Self::read_from_op_compressed(reader, &header).await;
+        }
+
+        Err(Error::new(
+            ErrorKind::InvalidResponse {
+                message: format!(
+                    "Invalid op code, expected {} or {} and got {}",
```

Comment:
Good point.  It is confusing because if none of the feature flags are enabled then the client is not expecting `OpCode::Compressed`, so we shouldn't say that it is expecting `OpCode::Compressed`.  Although the server will never send compressed messages if none of the feature flags are enabled (because the server only sends compressed messages if we tell it we can parse them).